src/Symfony/Component/Form/FormEvents.php

```diff
@@ -29,26 +29,45 @@ final class FormEvents
      *  - Change data from the request, before submitting the data to the form.
      *  - Add or remove form fields, before submitting the data to the form.
      *
+     * Any parent is being submitted during this event, it would have fired its PRE_SUBMIT
+     * event first, but its POST_SET_DATA status didn't changed yet.
+     *
+     * At this stage, a compound form hasn't submitted its children.
+     *
      * @Event("Symfony\Component\Form\FormEvent")
      */
     const PRE_SUBMIT = 'form.pre_bind';
 
     /**
-     * The SUBMIT event is dispatched just before the Form::submit() method
-     * transforms back the normalized data to the model and view data.
+     * The SUBMIT event is dispatched after the Form::submit() method
+     * has changed the view data by the request data, or submitted and mapped
+     * the children if the form is compound, and after reverse transformation
+     * to normalized representation.
+     *
+     * It's also dispatched just before the Form::submit() method transforms back
+     * the normalized data to the model and view data.
      *
-     * It can be used to change data from the normalized representation of the data.
+     * So at this stage children of compound forms are submitted and synchronized, unless
+     * their transformation failed, but a parent would still be at the PRE_SUBMIT level.
+     *
+     * Since the current form is not synchronized yet, it is still possible to add and
+     * remove fields.
      *
      * @Event("Symfony\Component\Form\FormEvent")
      */
     const SUBMIT = 'form.bind';
 
     /**
-     * The FormEvents::POST_SUBMIT event is dispatched after the Form::submit()
-     * once the model and view data have been denormalized.
+     * The FormEvents::POST_SUBMIT event is dispatched at the very end of the Form::submit().
+     *
+     * It this stage the model and view data may have been denormalized. Otherwise the form
+     * is desynchronized because transformation failed during submission.
      *
      * It can be used to fetch data after denormalization.
      *
+     * The event attaches the current view data, to know if this is the renormalized data
```

Comment:
Should this be 2 sentences?

The event attaches the current view data. To know if this is the...